### PR TITLE
[Feature] Display participants view data in the participants list

### PR DESCRIPTION
**Pull request by vhuseinova-msft**
## Purpose

* Display avatars and updated names for participants in the participants list (all data is saved and set locally)
* Renamed PersonaData to ParticipantViewData
* Added functionality to remove participant data when they leave the call

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
* Please turn on "Inject avatars" option in Settings

## What to Check
Verify that the following are valid
* images and names can be set for all (local and remote) participants in participants list 
* preset image and name are shown for a participant who has identifier's last symbol in 0...5 range

## Other Information

You may need to add a couple of users to the call to see avatars and updated names as some participants can have identifier's last symbol equal >5 or a letter.

### CODE REVIEW

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/Manager/AvatarViewManager.swift [hunk 1]
```diff
@@ -8,13 +8,14 @@ import AzureCommunicationCommon
...
+    func setRemoteParticipantViewData(for identifier: CommunicationIdentifier,
+                                      participantViewData: ParticipantViewData) -> Result<Void, Error>
 }
 
-class AvatarViewManager: AvatarViewManagerProtocol {
+class AvatarViewManager: AvatarViewManagerProtocol, ObservableObject {
     private let store: Store<AppState>
-    @Published private(set) var avatarStorage = MappedSequence<String, PersonaData>()
+    private(set) var avatarStorage = MappedSequence<String, ParticipantViewData>()
+    @Published var updatedId: String?
```
[the remark this answers is not in the document]
Comment (vhuseinova-msft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/Manager/AvatarViewManager.swift:18]: As discussed with @AmyL219, It's needed to be ObeservableObject for UIKIt to catch the changes from the avatarsStorage as for updating `ParticipantsListViewController` we need also update items list that is updated in `CompositeParticipantsList`

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/SwiftUI/Calling/Grid/ParticipantGridView.swift [hunk 2]
```diff
@@ -23,11 +23,11 @@ struct ParticipantGridView: View {
                 gridsCount = $0
             }
             .onReceive(viewModel.$displayedParticipantInfoModelArr) {
-                updateVideoViewManager(displayedRemoteInfoModelArr: $0)
+                updateVideoViewManagers(displayedRemoteInfoModelArr: $0)
             }
     }
 
-    func updateVideoViewManager(displayedRemoteInfoModelArr: [ParticipantInfoModel]) {
+    func updateVideoViewManagers(displayedRemoteInfoModelArr: [ParticipantInfoModel]) {
```
[the remark this answers is not in the document]
Comment (vhuseinova-msft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/SwiftUI/Calling/Grid/ParticipantGridView.swift]: This were updated during the testing, returned the correct name

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/SwiftUI/Calling/CallingViewComponent/InfoHeaderView.swift [hunk 3]
```diff
@@ -8,6 +8,7 @@ import FluentUI
 
 struct InfoHeaderView: View {
     @ObservedObject var viewModel: InfoHeaderViewModel
+    @ObservedObject var avatarViewManager: AvatarViewManager
```
Comment (AmyL219) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/SwiftUI/Calling/CallingViewComponent/InfoHeaderView.swift:11]: should we move the @observedObject to `CompositeParticipantsList`?

Comment (vhuseinova-msft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/SwiftUI/Calling/CallingViewComponent/InfoHeaderView.swift]: Updated!

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallCompositeOptions/CommunicationULocalDataOptions.swift [hunk 4]
```diff
@@ -0,0 +1,26 @@
+//
```
Comment (JoshuaLai) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/CallCompositeOptions/CommunicationULocalDataOptions.swift:1]: I think this file can be deleted. It is now replaced with the new `LocalSettings` file

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/FluentUI/Wrapper/CompositeParticipantsListCell.swift [hunk 5]
```diff
@@ -5,17 +5,22 @@
 
 import Foundation
 import FluentUI
+import UIKit
 
 class CompositeParticipantsListCell: TableViewCell {
 
     /// Set up the participant list item  in the participant list
     /// - Parameter viewModel: the participant view model needed to set up participant list cell
+    /// - Parameter avatarViewManager: the avatar view manager needed to set up avatar data
```
Comment (JoshuaLai) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/FluentUI/Wrapper/CompositeParticipantsListCell.swift:14]: nitpick: Not really necessary we only really comment block public apis

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/FluentUI/Wrapper/CompositeParticipantsListCell.swift [hunk 6]
```diff
@@ -5,17 +5,22 @@
...
     func setup(viewModel: ParticipantsListCellViewModel,
-               displayName: String) {
-        let isNameEmpty = viewModel.displayName.trimmingCharacters(in: .whitespaces).isEmpty
+               avatarViewManager: AvatarViewManager) {
+        let participantViewData = viewModel.getParticipantViewData(from: avatarViewManager)
+        let avatarParticipantName = viewModel.getParticipantName(with: participantViewData)
```
Comment (JoshuaLai) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/FluentUI/Wrapper/CompositeParticipantsListCell.swift:18]: I think this is a little hard to follow.
Why would we opt to use `getParticipantName` instead of `getCellDisplayName`? 
Do we have too many ways to get display name now?

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/Manager/RemoteParticipantsManager.swift [hunk 7]
```diff
@@ -33,19 +36,39 @@ class RemoteParticipantsManager: RemoteParticipantsManagerProtocol {
...
 
         participantsLastUpdateTimeStamp = state.remoteParticipantsState.lastUpdateTimeStamp
         let updatedParticipantsIds = Set(state.remoteParticipantsState.participantInfoList.map { $0.userIdentifier })
 
         let joinedParticipantsIds = updatedParticipantsIds.subtracting(participantsIds)
+        let leftParticipantsIds = participantsIds.subtracting(updatedParticipantsIds)
         participantsIds = updatedParticipantsIds
 
+        postRemoteParticipantsJoinedEvent(joinedParticipantsIds)
+        postRemoteParticipantsLeftEvent(leftParticipantsIds)
```
Comment (JoshuaLai) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/Manager/RemoteParticipantsManager.swift:51]: nitpick: I think we should use the term like `Removed` instead of `Left`

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/FluentUI/Wrapper/CompositeParticipantsListCell.swift (hunk 6 above)
Comment (vhuseinova-msft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/FluentUI/Wrapper/CompositeParticipantsListCell.swift:18]: For avatars images we use getParticipantName as it shouldn't use `(me)` inside it and for title - getCellDisplayName that contains the full title that should be shown to users. Accessibility also uses getCellDisplayName

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/FluentUI/Wrapper/CompositeParticipantsListCell.swift (hunk 5 above)
Comment (vhuseinova-msft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/FluentUI/Wrapper/CompositeParticipantsListCell.swift:14]: Just added avatarViewManager comment as the other info was there

**Review by AmyL219 — APPROVED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/Manager/RemoteParticipantsManager.swift (hunk 7 above)
Comment (vhuseinova-msft) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/AzureCommunicationUICalling/Presentation/Manager/RemoteParticipantsManager.swift]: Updated!

**Review by JoshuaLai — APPROVED** [review]
